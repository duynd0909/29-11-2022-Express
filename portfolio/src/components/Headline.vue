<template>
  <div class="headline">
    <div class="headline__primary">
      <div class="divider"></div>
      <span>{{ headline }}</span>
    </div>
    <slot name="sub-headline">
      <p class="headline__secondary">{{ subHealine }}</p>
    </slot>
  </div>
</template>
<script setup lang="ts"></script>
<script lang="ts">
export default {
  props: {
    headline: {
      type: String,
      required: true,
    },
    subHealine: {
      type: String,
      required: false,
      defualt: () => "",
    },
  },
};
</script>
<style lang="scss" scoped>
.headline {
  display: flex;
  flex-direction: column;
  width: fit-content;
  &__primary {
    color: var(--cinder-light);
    text-transform: uppercase;
    display: flex;
    align-items: center;
    gap: 16px;
    span {
      font-size: 12px;
      line-height: 24px;
      letter-spacing: 7.5px;
    }
    .divider {
      background: var(--gray-light);
      height: 2px;
      width: 16px;
    }
  }
  &__secondary {
    padding-top: 8px;
    font-family: $secondary-font;
    font-weight: 700;
    font-size: 36px;
    line-height: 45px;
    color: var(--cinder);
  }
}
</style>
