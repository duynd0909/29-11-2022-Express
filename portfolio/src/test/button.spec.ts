import {mount} from "@vue/test-utils";
import Headline from "../components/Headline.vue"
import { expect, test } from "vitest";
test("mount component",async()=>{
   expect(Headline).toBeTruthy();
   const wrapper = mount(Headline,{
       props:{
         headline:"aabc",
         subHeadline:"Ã¡B"
       },
   });
   expect(wrapper.element).toMatchSnapshot()
  })
