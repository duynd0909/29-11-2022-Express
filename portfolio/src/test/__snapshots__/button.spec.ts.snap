// Vitest Snapshot v1

exports[`mount component 1`] = `
<div
  class="headline"
  data-v-82f5b7be=""
  subheadline="Ã¡B"
>
  <div
    class="headline__primary"
    data-v-82f5b7be=""
  >
    <div
      class="divider"
      data-v-82f5b7be=""
    />
    <span
      data-v-82f5b7be=""
    >
      aabc
    </span>
  </div>
  
  <p
    class="headline__secondary"
    data-v-82f5b7be=""
  />
  
</div>
`;
